### PR TITLE
App insights: logback logging

### PR DESCRIPTION
**Pull request by doncem**
Automatically include logback event tracking for `uk.gov.hmcts` loggers.

```bash
13:25:10,661 |-INFO in AppenderAction - About to instantiate appender of type [com.microsoft.applicationinsights.logback.ApplicationInsightsAppender]
13:25:10,744 |-INFO in AppenderAction - Naming appender as [APP_INSIGHTS]
13:25:13,915 |-INFO in LoggerAction - Setting level of logger [uk.gov.hmcts] to INFO
```

### CODE REVIEW

File: java-logging-appinsights/src/main/resources/logback-includes.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<included>
+    <appender name="APP_INSIGHTS" class="com.microsoft.applicationinsights.logback.ApplicationInsightsAppender">
+        <instrumentationKey>${APPLICATION_INSIGHTS_IKEY:-00000000-0000-0000-0000-000000000000}</instrumentationKey>
```
Comment (timja-kainos) [inline java-logging-appinsights/src/main/resources/logback-includes.xml]: does it need a default?

File: README.md [hunk 2]
```diff
@@ -38,13 +38,13 @@ Maven:
 <dependency>
     <groupId>uk.gov.hmcts.reform</groupId>
     <artifactId>java-logging-spring</artifactId>
-    <version>1.5.0</version>
+    <version>1.5.1</version>
```
Comment (timja-kainos) [inline README.md]: 1.6.0?
seems like a new feature

Comment (doncem) [reply README.md]: extension of a feature. i keep on diminishing these kinda small additions to bugfix semver upgrade.

still want 1.6.0 though? just checking

File: java-logging-appinsights/src/main/resources/logback-includes.xml (hunk 1 above)
Comment (doncem) [reply java-logging-appinsights/src/main/resources/logback-includes.xml]: probably not. application will fail to start eitherway. if not here, than elsewhere. i'll remove default

File: README.md (hunk 2 above)
Comment (timja-kainos) [reply README.md]: I would prefer 1.6.0 its new functionality for sure.
but up to you

File: java-logging-appinsights/src/main/resources/logback-includes.xml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<included>
+    <appender name="APP_INSIGHTS" class="com.microsoft.applicationinsights.logback.ApplicationInsightsAppender">
```
Comment (timja-kainos) [inline java-logging-appinsights/src/main/resources/logback-includes.xml]: does this mean there needs to be a docs update to say you need to set your ROOT_APPENDER to APP_INSIGHTS?

File: java-logging-appinsights/src/main/resources/logback-includes.xml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (timja-kainos) [inline java-logging-appinsights/src/main/resources/logback-includes.xml]: does adding this here mean that other logback-includes won't be loaded?

File: java-logging-appinsights/src/main/resources/logback-includes.xml (hunk 3 above)
Comment (doncem) [reply java-logging-appinsights/src/main/resources/logback-includes.xml]: no, this appender is attached to **all** `uk.gov.hmcts` loggers

in certain situations, if appender reference is not found in the configuration (suppose you use APP_INSIGHTS in root) the application might fail to start. i think particularly if you temper with joran configuration manually.

- if added in root - there will be an error of missing APP_INSIGHTS appender
- if left like this - no error and you are free to use whichever java-logging library version and module. downside - hardcoded to `uk.gov.hmcts`

File: java-logging-appinsights/src/main/resources/logback-includes.xml (hunk 4 above)
Comment (doncem) [reply java-logging-appinsights/src/main/resources/logback-includes.xml]: good question. unfortunately, it gets overwritten

Comment (timja-kainos) [reply java-logging-appinsights/src/main/resources/logback-includes.xml]: would it be worth adding another optional include to the base logback.xml for this?

Comment (doncem) [reply java-logging-appinsights/src/main/resources/logback-includes.xml]: you're reading my mind 😃

**Pull request merged**